File: src/lessons/welcome/loopfor/LoopForEntity.c
//RemoteBuggle

void run(){
	/* BEGIN TEMPLATE */
	/* BEGIN SOLUTION */
	int cpt = 0;
	while (!isOverBaggle()) {
		cpt++;
		forward(1);
	}
	pickupBaggle();
	int cpt2;
	for (cpt2=0 ; cpt2<cpt ; cpt2++) {
		backward(1);
	}
	dropBaggle();
	/* END SOLUTION */
	/* END TEMPLATE */
}

File: src/lessons/welcome/loopfor/LoopForEntity.py
def run():
    global forward, backward
    _forward  = forward
    _backward = backward

    def forward(i=1):
        if i>1:
          errorMsg("Sorry Dave, I cannot let you use forward with an argument in this exercise. Use a loop instead.")
        else:
          _forward()
    def backward(i=1):
        if i>1:
          errorMsg("Sorry Dave, I cannot let you use backward with an argument in this exercise. Use a loop instead.")
        else:
          _backward()
    # BEGIN SOLUTION
    cpt = 0

    while not isOverBaggle():
      cpt = cpt+1
      forward()
    pickupBaggle()
    for i in range(cpt):
      backward()
    dropBaggle()
    # END SOLUTION
